src/installer/detector.ts

```diff
@@ -0,0 +1,66 @@
+import { getRelease, Release } from '@hashicorp/js-releases';
+import * as semver from 'semver';
+import * as vscode from 'vscode';
+import { exec } from '../utils';
+
+export const DEFAULT_LS_VERSION = 'latest';
+
+export function isValidVersionString(value: string): boolean {
+  return semver.validRange(value, { includePrerelease: true, loose: true }) !== null;
+}
+
+export async function getLsVersion(binPath: string): Promise<string | undefined> {
+  try {
+    const jsonCmd: { stdout: string } = await exec(binPath, ['version', '-json']);
+    const jsonOutput = JSON.parse(jsonCmd.stdout);
+    return jsonOutput.version;
+  } catch (err) {
+    // assume older version of LS which didn't have json flag
+    // return undefined as regex matching isn't useful here
+    // if it's old enough to not have the json version, we would be updating anyway
+    return undefined;
+  }
+}
+
+export async function getRequiredVersionRelease(
+  versionString: string,
+  extensionVersion: string,
+  vscodeVersion: string,
+): Promise<Release> {
+  const userAgent = `Terraform-VSCode/${extensionVersion} VSCode/${vscodeVersion}`;
+
+  // Take the user requested version and query the hashicorp release site
+  try {
+    const release = await getRelease('terraform-ls', versionString, userAgent);
+    console.log(`Found Terraform language server version ${release.version} which satisfies range '${versionString}'`);
+    return release;
+  } catch (err) {
+    if (versionString == DEFAULT_LS_VERSION) {
```

Comment:
```suggestion
    if (versionString === DEFAULT_LS_VERSION) {
```
We could add a eslint rule for checking this